src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicHelpers.cs

```diff
@@ -67,11 +67,7 @@ internal static unsafe T GetMsQuicParameter<T>(MsQuicSafeHandle handle, uint par
             &length,
             (byte*)&value);
 
-        if (StatusFailed(status))
-        {
-            throw ThrowHelper.GetExceptionForMsQuicStatus(status, $"GetParam({handle}, {parameter}) failed");
-        }
-
+        ThrowHelper.ThrowIfMsQuicError(status, $"GetParam({handle}, {parameter}) failed");
```

Comment:
This is now going to allocate a string for the error message even in the success case.

Reply:
I'm going to address this in some of my future PRs. Or @wfurt do you think you could revert these two in #88614?

Reply:
I think it is ok in interim as it is only perf optimization. I feel it is more important to get the API out.